kubernetes/base/stream/ws_client.py

```diff
@@ -179,7 +179,9 @@ def update(self, timeout=0):
         #                   efficient as epoll. Will work for fd numbers above 1024.
         # select.epoll()  - newest and most efficient way of polling.
         #                   However, only works on linux.
-        if sys.platform.startswith('linux') or sys.platform in ['darwin']:
+        #                   Will not work on Linux when eventlet is being used.
+        if ((sys.platform.startswith('linux') and getattr(select, "poll", None) is not None) or
```

Comment:
You can take inspiration from https://github.com/python/cpython/pull/21257 and https://github.com/ansible/ansible/pull/70352